scripts/migration/migrate_menas.py

```diff
@@ -0,0 +1,433 @@
+
+
+def get_distinct_entities_ids(db: Database, entity_names: List[str], collection_name: str, not_locked_only: bool,
+                              entity_name_field: str = "name", distinct_field: object = "name") -> List[str]:
+    """ General way to retrieve distinct entity field values (names, ids, ...) from non-locked entities """
+    collection = db[collection_name]
+    locked_filter = NOT_LOCKED_MONGO_FILTER if not_locked_only else EMPTY_MONGO_FILTER
+
+    entities = collection.distinct(
+        distinct_field,  # field to distinct on
+        {"$and": [
+            {entity_name_field: {"$in": entity_names}},  # filter on name (ds/mt)
+            locked_filter
+        ]}
+    )
+    return entities  # list of distinct names (in a single document)
+
+
+def get_distinct_mapping_tables_from_ds_names(db: Database, ds_names: List[str], not_locked_only: bool) -> List[str]:
+    ds_collection = db["dataset_v1"]
+    locked_filter = NOT_LOCKED_MONGO_FILTER if not_locked_only else EMPTY_MONGO_FILTER
+
+    mapping_table_names = ds_collection.aggregate([
+        {"$match": {"$and": [  # selection based on:
+            {"name": {"$in": ds_names}},  # dataset name
+            {"conformance": {"$elemMatch": {"_t": "MappingConformanceRule"}}},  # having some MCRs
+            locked_filter
+        ]}},
+        {"$unwind": "$conformance"},  # explodes each doc into multiple - each having single conformance rule
+        {"$match": {"conformance._t": "MappingConformanceRule"}},  # filtering only MCRs, other CR are irrelevant
+        {"$group": {
+            "_id": "notNeededButRequired",
+            "mts": {"$addToSet": "$conformance.mappingTable"}
+        }}  # grouping on fixed id (essentially distinct) and adding all MTs to a set
+    ])  # single doc with { _id: ... , "mts" : [mt1, mt2, ...]}
+
+    # if no MCRs are present, the result may be empty
+    mapping_table_names_list = list(mapping_table_names)  # cursor behaves one-iteration only.
+    if not list(mapping_table_names_list):
+        return []
+
+    extracted_list = mapping_table_names_list[0]['mts']
+    return extracted_list
+
+
+def assemble_notlocked_runs_from_ds_names(db: Database, ds_names: List[str]) -> List[str]:
+    return get_distinct_entities_ids(db, ds_names, "run_v1", entity_name_field="dataset", distinct_field="uniqueId",
+                                     not_locked_only=True)
+
+
+def assemble_schemas_from_ds_names(db: Database, ds_names: List[str], not_locked_only: bool) -> List[str]:
+    return assemble_schemas_from_x(db, ds_names, "dataset_v1", "schemaName", not_locked_only)
+
+
+def assemble_schemas_from_mt_names(db: Database, mt_names: List[str], not_locked_only: bool) -> List[str]:
+    return assemble_schemas_from_x(db, mt_names, "mapping_table_v1", "schemaName", not_locked_only)
+
+
+def assemble_schemas_from_x(db: Database, entity_names: List[str], collection_name: str,
```

Comment:
Not sure what the X is supposed to be

Reply:
this is a generalization of `assemble_schemas_from_ds_names` and `assemble_schemas_from_mt_names`, because their logic is very similar and that's why this common method exists.

Reply:
renamed to: `assemble_schemas`